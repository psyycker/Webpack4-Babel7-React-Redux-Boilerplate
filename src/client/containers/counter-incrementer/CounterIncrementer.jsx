import React from 'react';
import PropTypes from 'prop-types';

export default function CounterIncrementer(props) {
  const { counter, increaseCounter, decreaseCounter } = props;
  return (
    <div>
      <h1>Increase the counter through Redux :{counter}</h1>
      <input
        onClick={() => {
          increaseCounter();
        }}
        type="submit"
        value="increase"
      />
      <input
        onClick={() => {
          decreaseCounter();
        }}
        type="submit"
        value="decrease"
      />
    </div>
  );
}

CounterIncrementer.propTypes = {
  counter: PropTypes.number.isRequired,
  increaseCounter: PropTypes.func.isRequired,
  decreaseCounter: PropTypes.func.isRequired
};
