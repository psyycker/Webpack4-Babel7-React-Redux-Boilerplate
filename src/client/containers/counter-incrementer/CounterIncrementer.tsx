import React from 'react';
import PropTypes from 'prop-types';

export default class AppLayout extends React.Component {
  render() {
    const { counter, increaseCounter, decreaseCounter } = this.props;
    return (
      <div>
        <h1>
          Increase the counter through Redux :
          {counter}
        </h1>
        <input onClick={() => { increaseCounter(); }} type="submit" value="increase" />
        <input onClick={() => { decreaseCounter(); }} type="submit" value="decrease" />
      </div>
    );
  }
}

AppLayout.propTypes = {
  counter: PropTypes.number.isRequired,
  increaseCounter: PropTypes.func.isRequired,
  decreaseCounter: PropTypes.func.isRequired,
};
