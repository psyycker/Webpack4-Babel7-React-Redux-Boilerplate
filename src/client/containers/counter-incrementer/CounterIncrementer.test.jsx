import React from 'react';
import { render } from '@testing-library/react';
import CounterIncrementer from './CounterIncrementer';

test('Create counter page snapshot', () => {
  let value = 0;
  const increaseValue = () => {
    value += 1;
  };
  const { asFragment } = render(
    <CounterIncrementer
      counter={value}
      increaseCounter={increaseValue}
      decreaseCounter={() => {}}
    />
  );
  expect(asFragment()).toMatchSnapshot();
});
