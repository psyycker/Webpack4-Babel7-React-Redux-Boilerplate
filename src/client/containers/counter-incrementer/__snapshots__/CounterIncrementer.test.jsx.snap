// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create counter page snapshot 1`] = `
<div>
  <h1>
    Increase the counter through Redux :
    0
  </h1>
  <input
    onClick={[Function]}
    type="submit"
    value="increase"
  />
  <input
    onClick={[Function]}
    type="submit"
    value="decrease"
  />
</div>
`;
